test/functional/commands/find/by-css-e2e-specs.js

```diff
@@ -16,67 +16,72 @@ describe('Find - CSS', function () {
     await deleteSession();
   });
   it('should find an element by id (android resource-id)', async function () {
-    await driver.elementByCss('#text1').should.eventually.exist;
-    await driver.elementByCss('*[id="android:id/text1"]').should.eventually.exist;
-    await driver.elementByCss('*[resource-id="text1"]').should.eventually.exist;
+    // TODO: check the expected behavior when package is omitted. id in pagesrouce is 'andoird:id/text1' but
+    // the driver prepends a package name like 'io.appium.android.apis:id/text1'.
+    // await driver.$('#text1').elementId.should.eventually.exist;
```

Comment:
This line fails. Is this expected behavior or how should I fix?

Reply:
What about giving `android:` as well? `driver.$('#andoird:id/text1')`

Or you can set `disableIdLocatorAutocompletion` settings https://github.com/appium/appium-uiautomator2-driver/#settings-api, then maybe driver.$('#andoird:id/text1')` will work

Reply:
Sorry I realized `#text1` is valid css selector value as I confirmed css-converter can handle it.
And the reason why this test fails is because same reason as https://github.com/appium/appium-uiautomator2-driver/pull/578#discussion_r1093244242

the driver automatically prepends the package name and sends a request to uia2 server like this way
```
Proxying [POST /element] to [POST http://127.0.0.1:8200/session/3a22c73d-af92-44a0-8e66-c6b350c8320f/element] with body: {"strategy":"-android uiautomator","selector":"new UiSelector().resourceId(\"io.appium.android.apis:id/text1\")","context":"","multiple":false}
```
but the resource-id is `android:id/text1`. So I tried `#android:id/text1` but it doesn't work because CssConverter assumes it's invalid as a locator.

My question is why do both the driver and the server need to care about the package name?
driver: https://github.com/appium/appium-uiautomator2-driver/blob/c88658da1897729e7e3856a88cb9314212eb1716/lib/css-converter.js#L160
server: https://github.com/appium/appium-uiautomator2-server/blob/12866ed644c68bec53cadf6599490a0619b41938/app/src/main/java/io/appium/uiautomator2/utils/ElementLocationHelpers.java#L95

Reply:
>  removing the driver's side would solve this confusion

ah just removing it won't save the `android:id/text1` case.

Reply:
I think you still can provide `android:id/text1` as identifier, but the colon and the slash must be escaped: `#android\\:id\\/identifier`

Reply:
wow it works thank you!

Reply:
I fixed it https://github.com/appium/appium-uiautomator2-driver/pull/578/commits/0432006e3893cc5306405d6b6ae1fc5b777019b1
but I wonder if it's fine to change because I think the intention of the original test case was to see if the omitted package part in the locator is automatically completed 🤔

Reply:
The change existed over 2 years, so maybe the test must have been updated. We did not get any issue report as well, so probably it would be good to follow current implementation (so needed to update test case before)